pyccel/parser/semantic.py

```diff
@@ -1869,7 +1867,15 @@ def _visit_Assign(self, expr, **settings):
 
         for l, r in zip(lhs,rhs):
             is_pointer_i = l.is_pointer if isinstance(l, (Variable, DottedVariable)) else is_pointer
-
+            if isinstance(l, Variable) and l.is_ndarray:
+                if l.is_target:
+                    errors.report(REASSIGN_TARGET,
+                          bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
+                          severity='error', symbol=l.name)
+                if isinstance(r, Variable) and r.allocatable and l.allocatable:
+                    errors.report(ASSIGN_ALLOCATABLES,
+                          bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
+                          severity='error', symbol=l.name)
             new_expr = Assign(l, r)
```

Comment:
Are you sure this is the right place for this code? I suspect any modification of the properties has already been done by this point. I expected to see this code in `_ensure_target` or `_assign_lhs_variable`

Reply:
from what i tested i have all the properties that is needed by this point. 
but I will try to move them to what you suggested I makes more sense to put them there if I have all the properties.

Reply:
It looks like it's in the right place now :+1: